SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLatencyTracer.cpp

```diff
@@ -36,15 +36,15 @@ class UEStream : public std::stringbuf
 UEStream UStream;
 
 #if TRACE_LIB_ACTIVE
-improbable::trace::SpanContext ReadSpanContext(const void* TraceBytes, const void* SpanBytes)
+const improbable::trace::SpanContext ReadSpanContext(const void* TraceBytes, const void* SpanBytes)
```

Comment:
Nit: Returning by const value doesn't prevent you from copying into a non-const variable, so I'd advise against ever returning by const value (perhaps there is a valid use case but I'm not aware of any).

Reply:
This was done to work around SpanContext having it's move constructor deleted in latest trace library packages

Reply:
Could you expand on this? I'm confused as to how these can affect each other and would normally agree that returning something by const value is odd.

Reply:
Returning by non-const value invoked the move constructor, whereas switching it to const return doesn't 🤷

Reply:
Took a look at the trace library and realised I'm being and idiot. I read this as the constructors had been removed instead of set to `= delete`.
This is probably an oversight in the worker SDK and we should raise it as a bug. What's happening here is that it can't elide the result for some reason, so it tries to move return it. But the overload is deleted so that's an error. Using a const value makes copying the only viable overload so it compiles. 

I suspect that what's intended is that there isn't an "efficient" move, however in that case ideally the move constructor is either not defined at all, set explicitly to call the copy constructor/assignment.

Reply:
Thanks for raising this with the worker team. This can be resolved when either we remove the latency tracer (deprecated in favour of event tracing), or worker roll out a fix.